src/charm.py

```diff
@@ -57,66 +56,72 @@ def __init__(self, *args) -> None:
         self._certificates = TLSCertificatesRequiresV3(self, "certificates")
 
         self.framework.observe(self.on.config_changed, self._configure_nssf)
+        self.framework.observe(self.on.update_status, self._configure_nssf)
         self.framework.observe(self.on.nssf_pebble_ready, self._configure_nssf)
         self.framework.observe(self.on.fiveg_nrf_relation_joined, self._configure_nssf)
         self.framework.observe(self._nrf_requires.on.nrf_available, self._configure_nssf)
         self.framework.observe(self._nrf_requires.on.nrf_broken, self._on_nrf_broken)
-        self.framework.observe(
-            self.on.certificates_relation_created, self._on_certificates_relation_created
-        )
-        self.framework.observe(
-            self.on.certificates_relation_joined, self._on_certificates_relation_joined
-        )
+        self.framework.observe(self.on.certificates_relation_joined, self._configure_nssf)
         self.framework.observe(
             self.on.certificates_relation_broken, self._on_certificates_relation_broken
         )
-        self.framework.observe(
-            self._certificates.on.certificate_available, self._on_certificate_available
-        )
+        self.framework.observe(self._certificates.on.certificate_available, self._configure_nssf)
         self.framework.observe(
             self._certificates.on.certificate_expiring, self._on_certificate_expiring
         )
 
-    def _configure_nssf(
-        self,
-        event: EventBase,
-    ) -> None:
-        """Configure NSSF configuration file and pebble service.
-
-        Args:
-            event (EventBase): Juju event
-        """
+    def ready_to_configure(self) -> bool:
+        """Returns whether all preconditions are met to proceed with configuration."""
```

Comment:
Similar to my other comment, I'm not convinced about this method doing 2 different things and the method name not matching with the most important thing this method does (setting statuses). I'd recommend only have it deal with statuses and do it all the way (also take care of the active status). I.e. more or less what we are doing for check status elsewhere.